modules/utils/include/hephaestus/utils/exception.h

```diff
@@ -107,6 +107,14 @@ class InvalidOperationException : public heph::Exception {
   }
 };
 
+//=================================================================================================
+/// Exception raised due to a hardware issue
+class HardwareIssueException : public heph::Exception {
+public:
+  HardwareIssueException(const std::string& msg, std::source_location loc) : Exception(msg, loc) {
```

Comment:
Terms `Issue` and `Exception` seem to be redundant. Suggestion: `HardwareException`.

Reply:
True, fixed.